### PR TITLE
Feature/ipython

### LINKED ISSUE #1375: Load bokehjs in a notebook during `import bokeh`
Testing for notebook is cheap, so we could just `load_notebook()` in `bokeh.__init__` and fail silently if not available. That would allow us to skip writing this: 
```
from bokeh import load_notebook
load_notebook()
```
in every notebook.

### PR DESCRIPTION
**Pull request by damianavila**
This PR closes #1375 and indirectly #1697 and #867.

IPython/Jupyter does not provide a way to know if we are inside a notebook for multiples reasons already exposed in mailing list and other places. Essentially, the frontend does not know about the things happening at the kernel level. So, to avoid some hacks prone to give errors, let's just load the notebook stuff at the `bokeh.__init__` and fails silently if IPython is not installed. Also provide a way to customize the load_notebook thought some `bokeh.settings` and finally fix all the examples and documentation which does not need to import the `load_notebook` anymore. This also fix, as a side effect the fails importing load_notebook at the IPython/Jupyter 3.0-dev/master...

### CODE REVIEW

File: bokeh/__init__.py [hunk 1]
```diff
@@ -10,10 +14,9 @@
     __version__ = get_versions()['version']
     del get_versions
 
-
 _notebook_loaded = None
 
-def load_notebook(resources=None, verbose=False, force=False, skip=False):
+def load_notebook(resources=None, verbose=False, skip=False):
```
Comment (bryevdv) [inline bokeh/__init__.py]: `skip` doesn't actually appear to get used? Maybe use `skip` below to omit call to  `load_notebook` altogether, and remove it as a param?

**Comment by bryevdv** [comment]
One small comment, otherwise LGTM

File: bokeh/__init__.py (hunk 1 above)
Comment (damianavila) [inline bokeh/__init__.py]: but skip is used here: https://github.com/bokeh/bokeh/blob/master/bokeh/__init__.py#L82
or you mean it is not frequently used??

Comment (bryevdv) [inline bokeh/__init__.py]: Oh I see I forget the usage be outside the diff. It's not often used, but it should also be renamed. Can you change it to `show_banner` or something similar (and update the template as well)? And then `skip` could be used to skip loading in `__init__.py` (might be sometimes useful)

Comment (damianavila) [inline bokeh/__init__.py]: ok... I will do it later today or tomorrow morning...

**Comment by damianavila** [comment]
OK, latest requests about the `hide_banner` and `skip_load` pushed... ready to another look...

**Comment by bryevdv** [comment]
LGTM

**Comment by mattpap** [comment]
One thing that was overlooked is that "import bokeh" won't work without static files (causes `IOError`).

**Comment by damianavila** [comment]
>One thing that was overlooked is that "import bokeh" won't work without static files (causes IOError).

I miss this... you suggest just caching the IOerror and pass or do you want some warning?

**Comment by bryevdv** [comment]
`log.debug` seems reasonable

**Pull request merged**